File: 104. Maximum Depth of Binary Tree.swift
public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?
    public init(_ val: Int) {
        self.val = val
        self.left = nil
        self.right = nil
    }
}

class Solution {
    var maxdepth = 0
    
    func maxDepth(_ root: TreeNode?) -> Int {
        helper(root, depthLevel: 0)
        return maxdepth
    }
    
    func helper(_ root: TreeNode?, depthLevel: Int) {
        
        guard let node = root else {
            return
        }
        
        let currentLevel = depthLevel + 1
        
        if node.left == nil && node.right == nil {
            maxdepth = max(maxdepth, currentLevel)
            return
        }
        
        if node.left != nil {
            helper(node.left, depthLevel: currentLevel)
        }
        
        if node.right != nil {
            helper(node.right, depthLevel: currentLevel)
        }
    }
}

File: LeetCode/104.Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.trace(root, 0)
        
    def trace(self, node, num):
        if node == None:
            return num
        return max(self.trace(node.left, num + 1), self.trace(node.right, num + 1))
